src/mono/mono/mini/method-to-ir.c

```diff
@@ -6452,8 +6452,10 @@ mono_method_to_ir (MonoCompile *cfg, MonoMethod *method, MonoBasicBlock *start_b
 		generic_context = &generic_container->context;
 	cfg->generic_context = generic_context;
 
-	if (!cfg->gshared)
-		g_assert (!sig->has_type_parameters);
+	if (!cfg->gshared) {
+		if (method->wrapper_type != MONO_WRAPPER_OTHER)
+			g_assert (!sig->has_type_parameters);
+	}
```

Comment:
I don't really like this.  It's risky for two reasons:
1. There's a ton of `MONO_WRAPPER_OTHER` subtypes.  Can you modify the change to skip only the unsafe accessor wrapper subtype?
2. The AOT compiler probably depends on not seeing a type parameter in a ton of other places in the compiler - skipping this assertion you're just getting lucky that nothing else blows up.

I think really we need the AOT compiler to just skip these kinds of generic method since they're invalid and we don't expect to really see them in net8.0.  I'm not sure how to do that.  If we're targeting this fix for net8.0, I'm ok with just skipping the assertion only for the unsafe accessor wrapper subtype.

Reply:
AOT compile still needs to generate code for the Unsafe Accessor wrappers in full AOT mode, doesn't it?

Reply:
Yea that's true. But I'm not sure if it's going to generate something that is callable, to be honest.